static/js/read_receipts.js

```diff
@@ -0,0 +1,54 @@
+import $ from "jquery";
+import SimpleBar from "simplebar";
+
+import render_read_receipts from "../templates/read_receipts.hbs";
+import render_read_receipts_modal from "../templates/read_receipts_modal.hbs";
+
+import * as channel from "./channel";
+import {$t} from "./i18n";
+import * as loading from "./loading";
+import * as message_store from "./message_store";
+import * as overlays from "./overlays";
+import * as people from "./people";
+
+export function show_user_list(message_id) {
+    $("body").append(render_read_receipts_modal());
+    overlays.open_modal("read_receipts_modal", {
+        autoremove: true,
+        micromodal: true,
+        on_show: () => {
+            loading.make_indicator($("#read_receipts_modal .loading_indicator"));
+            channel.get({
+                url: `/json/messages/${message_id}/read_receipts`,
+                idempotent: true,
+                success(data) {
+                    const message_author_id = message_store.get(message_id).sender_id;
+                    const user_ids = data.user_ids.filter((id) => id !== message_author_id); // Do not show author.
+
+                    let users = user_ids.map((id) => people.get_by_user_id(id));
+                    users = users.filter((user) => !user.is_bot);
+                    users = users.map((user) => ({
+                        full_name: user.full_name,
+                        avatar_url: user.avatar_url,
+                    }));
+                    users.sort(people.compare_by_name);
+
+                    loading.destroy_indicator($("#read_receipts_modal .loading_indicator"));
+                    if (users.length === 0) {
+                        $("#read_receipts_modal .read_receipts_info").text(
+                            $t({defaultMessage: "No one has read this message yet."}),
+                        );
+                    } else {
+                        $("#read_receipts_modal .modal__container").addClass(
+                            "showing_read_receipts_list",
+                        );
+                        $("#read_receipts_modal .modal__content").append(
+                            render_read_receipts({users}),
+                        );
+                        new SimpleBar($("#read_receipts_modal .modal__content")[0]);
```

Comment:
What's the thinking behind doing this vs. having a `data-simplebar` declaration in the modal itself?

Reply:
We are adding the read receipts list after the modal has been created. Using `data-simplebar` declaration in the modal itself would render all the simplebar specific wrappers before we fetch and render read receipts.

Using `.append` to add read receipts list in the above code is adding the list as a sibling element to the simplebar wrapper. We might work around that with help of some CSS selectors specific to simplebar but I'm afraid they might change later. And this feels a bit clean.

Reply:
OK I guess the point is that we're rendering an empty modal while waiting for the network response. I think that's OK, but we should use `loading.js` to add a loading indicator while waiting for the server to return the data.

@ganpa3 or @sahil839 can help if you need it, but I think it's a pretty simple library.

Reply:
We [have a loading indicator](https://github.com/zulip/zulip/pull/18935/commits/42b86dd522f2f92fb3f3bb835479389294141f96#diff-2a478878505fa314bc9821ae5e3b5ac0bdaccf1b76ee3864ac9b69905858d3daR19) already. Do you mean something else? :thinking:

Reply:
OK, thanks for confirming :)

Reply:
It appears like this while it waits for fetching read receipts

![image](https://user-images.githubusercontent.com/44665669/164953563-a041b611-a6ba-4fcf-a2fe-7323e395b88b.png)

Reply:
I guess it doesn't make much sense to show "This message has been...." line especially now that we want to show number of people who read the message. Not it'd just look like 
![image](https://user-images.githubusercontent.com/44665669/164954303-67ef8d6b-3c45-4d1e-a1fd-72e358b72aee.png)

Reply:
Yeah, the second screenshot is much nicer.